bin/build_ztunnel.sh

```diff
@@ -115,6 +115,12 @@ function maybe_build_ztunnel() {
 
 # ztunnel binary vars (TODO handle debug builds, centos, arm, darwin etc.)
 ISTIO_ZTUNNEL_BASE_URL="${ISTIO_ZTUNNEL_BASE_URL:-https://storage.googleapis.com/istio-build/ztunnel}"
+
+# If we are not using the default, assume its private and we need to authenticate
+if [[ "${ISTIO_ZTUNNEL_BASE_URL}" != "https://storage.googleapis.com/istio-build/ztunnel" ]]; then
```

Comment:
I don't think this check is necessary, and for other vendors building istio this will present a problem because our base url will be different.
+ init.sh

Reply:
For other vendors it will mean you _do_ authenticate, which is probably what you want?

This just avoids the public jobs or developers authenticating when they don't need to which may cause issues (don't have gcloud, etc)